### PR TITLE
base: add thrift dependencies

### PR DESCRIPTION
**Pull request by cfriedt**
Add the thrift compiler (code generator).

### CODE REVIEW

File: Dockerfile.base [hunk 1]
```diff
@@ -78,7 +78,10 @@ RUN apt-get -y update && \
 		valgrind \
 		wget \
 		ovmf \
-		xz-utils
+		xz-utils \
+		libboost-all-dev \
+		thrift-compiler \
+		libthrift-dev
```
Comment (cfriedt) [inline Dockerfile.base:84]: Technically speaking, only `thrift-compiler` is required for Zephyr builds. The other two are only required for testing against native applications.

Comment (cfriedt) [reply Dockerfile.base:84]: @stephanosio - what is your preference for the container? To add only thrift-compiler?

Comment (cfriedt) [reply Dockerfile.base]: I removed the additional packages, so only `thrift-compiler` is pulled in.

Comment (cfriedt) [reply Dockerfile.base]: Now that I think about it, I think `libthrift-dev` might still be necessary.

File: Dockerfile.base [hunk 2]
```diff
@@ -78,7 +78,8 @@ RUN apt-get -y update && \
 		valgrind \
 		wget \
 		ovmf \
-		xz-utils
+		xz-utils \
+		thrift-compiler
```
Comment (stephanosio) [inline Dockerfile.base:82]: Is this all that is required to build Thrift tests and samples in the CI?

Comment (cfriedt) [reply Dockerfile.base:82]: Yes. If the user wants to run the host-side demos, then there are additional things to install, but they are not required for simply building the Zephyr tests.

**Review by stephanosio — APPROVED** [review]

**Pull request merged**